src/ckeditor/signature/InsertSignatureCommand.js

```diff
@@ -54,24 +61,47 @@ export default class InsertSignatureCommand extends Command {
 		// which has a loosened schema.
 		const modelFragment = editor.data.toModel(viewFragment)
 
-		const signature = writer.createElement('signature')
-		writer.append(writer.createText('--'), signature)
-		writer.append(writer.createElement('paragraph'), signature)
-		writer.append(modelFragment, signature)
+		const signatureElement = writer.createElement('signature')
+		writer.append(writer.createText('-- '), signatureElement)
+		writer.append(writer.createElement('paragraph'), signatureElement)
+		writer.append(modelFragment, signatureElement)
+		if (signatureAboveQuote) {
+			writer.append(writer.createElement('paragraph'), signatureElement)
+		}
+
+		const signaturePosition = signatureAboveQuote ? this.findPositionAboveQuote(editor, writer) : writer.createPositionAt(editor.model.document.getRoot(), 'end')
+		editor.model.insertContent(signatureElement, signaturePosition)
+	}
 
-		editor.model.insertContent(
-			signature,
-			writer.createPositionAt(editor.model.document.getRoot(), 'end')
+	/**
+	 *
+	 * @param {*} editor the editor instance
+	 * @param {*} writer the writer instance
+	 * @returns {*} the position above the quoted text; position 1 if no quote found
+	 */
+	findPositionAboveQuote(editor, writer) {
+		// Create a range spanning over the entire root content:
+		const range = editor.model.createRangeIn(
+			editor.model.document.getRoot()
 		)
+
+		// Iterate over all items in this range:
+		for (const value of range.getWalker({ shallow: true })) {
+			if (value.item.is('element') && value.item.name === 'quote') {
+				return writer.createPositionBefore(value.item)
+			}
+		}
+
+		return writer.createPositionAt(editor.model.document.getRoot(), 1)
```

Comment:
```suggestion
		return writer.createPositionAt(editor.model.document.getRoot(), 'end')
```

If we reach this point there is no quote right? We should append the signature to the end then.

Reply:
yes, sounds reasonable